### DISCUSSION TITLE
rdf4j and spring-boot 3.x

### DISCUSSION THREAD

**Discussion by naturzukunft** [eclipse-rdf4j/rdf4j#4611] [General]
Hi there,
currently i try to upgrade my application to spring-boot 3.x.
And i have trouble with 
```
		<dependency>
		    <groupId>org.eclipse.rdf4j</groupId>
		    <artifactId>rdf4j-spring-boot-sparql-web</artifactId>
		    <version>4.3.1</version>
		</dependency>
```
-> ` java.lang.ClassNotFoundException: jakarta.servlet.http.HttpSessionContext`

Are there plans to switch to a new version with rdf4j 5.x ?

Fredy

**Comment by hmottestad** [comment]
@erikgb has been working on it and made decent progress.

**Reply by erikgb** [reply]
This is the next step: https://github.com/eclipse/rdf4j/pull/4604, but still far from Spring Boot 3 compatible.

The main blocker IMO is the Solr/ElasticSearch/Lucene transitive dependencies, in particular legacy Jetty. I am probably not capable of fixing that, but I hope @barthanssens eventually can take a look.

**Comment by barthanssens** [comment]
5.x would indeed be an opportunity to move to the latest and greatest ES/Solr (probably breaking stuff, IIRC some geo-features in ES moved to different packages and/or changes in method signatures)

**Reply by erikgb** [reply]
Seems like at least Solr is still a blocker for Spring Boot 3. No available version containing the javax -> jakarta migration. Ref. https://issues.apache.org/jira/plugins/servlet/mobile#issue/SOLR-16441

**Comment by naturzukunft** [comment]
Hi there!
One year later i run into this problem again!

Are there any news? Plans, etc.

Fredy

**Reply by hmottestad** [reply]
It's a very big and complex task. @erikgb started on it, but there are just so many interdependencies in the project that a lot of things need to be updated at the same time. I started moving us away from all the old elasticsearch dependencies, which would free up Lucene and Solr upgrades, but there is really a lot of code that needs to be rewritten: https://github.com/eclipse-rdf4j/rdf4j/pull/4777

**Reply by naturzukunft** [reply]
That's simple in my case. it's just `org.eclipse.rdf4j.rdf4j-spring-boot-sparql-web`.

**Reply by hmottestad** [reply]
Yeah. Sorry. I realised what you meant right after  I had submitted my comment. So I quickly removed it hoping you wouldn't notice my stupid question 😂

**Comment by dennisvang** [comment]
also see:
- #5063
